lib/controllers/object.js

```diff
@@ -225,6 +226,62 @@ exports.getObjectAcl = async function getObjectAcl(ctx) {
   };
 };
 
+
+/**
+ * POST Object
+ * Post an object into a bucket
+ * {@link https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html}
+ */
+exports.postObject = async function postObject(ctx) {
+
+  try {
+    const [fields, files] = await _promisifyUpload(ctx.req);
+
+    const object = new S3Object(
+      ctx.params.bucket,
+      fields['key'][0],
+      files['file'][0],
+      fields
+    );
+
+    const md5Hash = await ctx.store.postObject(object);
+
+    ctx.type = "application/xml";
+    ctx.status = 201;
+    ctx.body = {
+      PostResponse: {
+        Etag: md5Hash,
+        Bucket: ctx.params.bucket,
+        Key: object.key,
+        Location: ctx.href.split("?")[0] + '/' + object.key
+      }
+    };
+
+    ctx.logger.info(
+      'Stored object "%s" in bucket "%s" successfully',
+      object.key,
+      object.bucket
+    );
+  } catch (err) {
+    ctx.logger.error(
+      'Error uploading object to bucket "%s"',
+      ctx.params.bucket,
+      err
+    );
+    throw err;
+  }
+};
+
+const _promisifyUpload = (req) => new Promise((resolve, reject) => {
```

Comment:
You can just have this function inlined.

Reply:
done